sdk/core/src/context.rs

```diff
@@ -6,15 +10,84 @@
 /// pipeline execution history between policies.
 /// For example, it could be used to signal that an execution failed because a CosmosDB endpoint is
 /// down and the appropriate policy should try the next one).
-#[derive(Clone)]
+#[derive(Clone, Debug, Default)]
 pub struct Context {
-    // Temporary hack to make sure that Context is not initializeable
-    // Soon Context will have proper data fields
-    _priv: (),
+    property_bag: HashMap<TypeId, Arc<dyn Any + Send + Sync>>,
```

Comment:
I'm confused why we feel the need for a loosely typed property bag when we can just have an optional field for the mock transaction.

Reply:
An SDK user should be able to implement their own pipeline policy. We need to give them a way to pass context information  to their policy. With the property bag they can (in a type safe way too!).
If we were to hardcode only the fields useful to us custom policies would not have a way to pass information in the context.